js/battle.js

```diff
@@ -3219,7 +3219,16 @@ var Battle = (function () {
 			if (kwargs.prepare || kwargs.anim === 'prepare') {
 				this.prepareMove(pokemon, move, target);
 			} else if (!kwargs.notarget) {
-				(kwargs.anim ? Tools.getMove(kwargs.anim) : move).anim(this, [pokemon.sprite, target.sprite]);
+				var usedMove = kwargs.anim ? Tools.getMove(kwargs.anim) : move;
+				if (kwargs.spread) {
+					var targets = [pokemon.sprite].concat(target.side.active.map(function (active) { return active.sprite; }));
+					if (target.side.active.length > 2 && target.side.active.indexOf(target) !== 1) {
+						targets.splice((target.side.active.indexOf(target) ? 1 : 3), 1);
```

Comment:
`.splice` is the entirely wrong way to do it, it's really slow. Just construct it with `.push`